### ISSUE TITLE
brew install on mac air fails

### ISSUE THREAD

**Issue by mkinney** [NASA-SW-VnV/homebrew-core#4]
Looks like the it will not install on a mac air.

```
% brew install nasa-sw-vnv/core/ikos
llvm@9: The x86_64 architecture is required for this software.
Error: ikos: An unsatisfied requirement failed this build.
```

**Comment by CIPop** [comment]
I was able to install on M1 but running it fails.

```
ikos test.c
[*] Compiling test.c
[*] Running ikos preprocessor
[*] Running ikos analyzer
[*] Translating LLVM bitcode to AR
[*] Running liveness analysis
[*] Running widening hint analysis
[*] Running interprocedural value analysis
[*] Analyzing entry point 'main'
[*] Checking properties for entry point 'main'
PLEASE submit a bug report to https://github.com/Homebrew/homebrew-core/issues and include the crash backtrace.
Stack dump:
0.	Program arguments: /opt/homebrew/Cellar/ikos/3.1_1/bin/ikos-analyzer -a=fca,boa,nullity,dbg,uva,shc,prover,dca,dfa,pcmp,dbz,sio,sound -d=interval -entry-points=main -globals-init=skip-big-arrays -proc=inter -j=1 -widening-strategy=widen -widening-delay=1 -widening-period=1 -narrowing-strategy=narrow -allow-dbg-mismatch -display-checks=no -display-inv=no -log=info -progress=auto /var/folders/1x/rd5lvrvs3_z6v_wtfvtspjt00000gn/T/ikos-cm6bcuux/test.pp.bc -o output.db
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  ikos-analyzer            0x000000010094ba24 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
1  ikos-analyzer            0x000000010094bdf8 SignalHandler(int) + 188
2  libsystem_platform.dylib 0x000000018e8a7a24 _sigtramp + 56
3  libunwind.dylib          0x000000019b097b2c _Unwind_GetIP + 212
4  libc++abi.dylib          0x000000018e832f40 __gxx_personality_v0 + 296
5  libunwind.1.0.dylib      0x000000010197c928 unwind_phase2 + 372
6  libunwind.1.0.dylib      0x000000010197cb30 _Unwind_Resume + 92
7  ikos-analyzer            0x0000000100589400 ikos::core::machine_int::Congruence::singleton() const + 0
8  ikos-analyzer            0x0000000100589130 std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> ikos::core::patricia_tree_map_impl::update_or_insert<ikos::analyzer::Variable*, ikos::core::machine_int::Interval, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&)>(std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> const&, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&) const&, ikos::analyzer::Variable* const&, ikos::core::machine_int::Interval const&) + 944
9  ikos-analyzer            0x0000000100589130 std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> ikos::core::patricia_tree_map_impl::update_or_insert<ikos::analyzer::Variable*, ikos::core::machine_int::Interval, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&)>(std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> const&, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&) const&, ikos::analyzer::Variable* const&, ikos::core::machine_int::Interval const&) + 944
10 ikos-analyzer            0x0000000100589130 std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> ikos::core::patricia_tree_map_impl::update_or_insert<ikos::analyzer::Variable*, ikos::core::machine_int::Interval, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&)>(std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> const&, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&) const&, ikos::analyzer::Variable* const&, ikos::core::machine_int::Interval const&) + 944
11 ikos-analyzer            0x0000000100588ef0 std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> ikos::core::patricia_tree_map_impl::update_or_insert<ikos::analyzer::Variable*, ikos::core::machine_int::Interval, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&)>(std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> const&, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&) const&, ikos::analyzer::Variable* const&, ikos::core::machine_int::Interval const&) + 368
12 ikos-analyzer            0x0000000100588ef0 std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> ikos::core::patricia_tree_map_impl::update_or_insert<ikos::analyzer::Variable*, ikos::core::machine_int::Interval, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&)>(std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> const&, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&) const&, ikos::analyzer::Variable* const&, ikos::core::machine_int::Interval const&) + 368
13 ikos-analyzer            0x0000000100589130 std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> ikos::core::patricia_tree_map_impl::update_or_insert<ikos::analyzer::Variable*, ikos::core::machine_int::Interval, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&)>(std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> const&, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&) const&, ikos::analyzer::Variable* const&, ikos::core::machine_int::Interval const&) + 944
14 ikos-analyzer            0x0000000100589130 std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> ikos::core::patricia_tree_map_impl::update_or_insert<ikos::analyzer::Variable*, ikos::core::machine_int::Interval, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&)>(std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> const&, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&) const&, ikos::analyzer::Variable* const&, ikos::core::machine_int::Interval const&) + 944
15 ikos-analyzer            0x0000000100589130 std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> ikos::core::patricia_tree_map_impl::update_or_insert<ikos::analyzer::Variable*, ikos::core::machine_int::Interval, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&)>(std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> const&, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&) const&, ikos::analyzer::Variable* const&, ikos::core::machine_int::Interval const&) + 944
16 ikos-analyzer            0x0000000100589130 std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> ikos::core::patricia_tree_map_impl::update_or_insert<ikos::analyzer::Variable*, ikos::core::machine_int::Interval, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&)>(std::__1::shared_ptr<ikos::core::patricia_tree_map_impl::PatriciaTree<ikos::analyzer::Variable*, ikos::core::machine_int::Interval> const> const&, ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&) const&, ikos::analyzer::Variable* const&, ikos::core::machine_int::Interval const&) + 944
17 ikos-analyzer            0x0000000100588d34 void ikos::core::PatriciaTreeMap<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::update_or_insert<ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&)>(ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&)::'lambda'(ikos::core::machine_int::Interval const&, ikos::core::machine_int::Interval const&) const&, ikos::analyzer::Variable* const&, ikos::core::machine_int::Interval const&) + 28
18 ikos-analyzer            0x0000000100588bac ikos::core::machine_int::SeparateDomain<ikos::analyzer::Variable*, ikos::core::machine_int::Interval>::refine(ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&) + 152
19 ikos-analyzer            0x000000010057f644 ikos::core::machine_int::IntervalDomain<ikos::analyzer::Variable*>::add(ikos::core::machine_int::Predicate, ikos::analyzer::Variable*, ikos::analyzer::Variable*) + 940
20 ikos-analyzer            0x000000010057f494 ikos::core::machine_int::IntervalDomain<ikos::analyzer::Variable*>::add(ikos::core::machine_int::Predicate, ikos::analyzer::Variable*, ikos::analyzer::Variable*) + 508
21 ikos-analyzer            0x00000001005a96ac ikos::core::scalar::CompositeDomain<ikos::analyzer::Variable*, ikos::analyzer::MemoryLocation*, ikos::core::uninitialized::SeparateDomain<ikos::analyzer::Variable*>, ikos::core::machine_int::PolymorphicDomain<ikos::analyzer::Variable*>, ikos::core::nullity::SeparateDomain<ikos::analyzer::Variable*> >::int_add(ikos::core::machine_int::Predicate, ikos::analyzer::Variable*, ikos::analyzer::Variable*) + 176
22 ikos-analyzer            0x000000010073b19c ikos::analyzer::BufferOverflowChecker::check_memory_location_access(ikos::ar::Statement*, ikos::ar::Value*, ikos::ar::Value*, ikos::core::exception::ExceptionDomain<ikos::core::memory::PolymorphicDomain<ikos::analyzer::Variable*, ikos::analyzer::MemoryLocation*> > const&, ikos::analyzer::MemoryLocation*, ikos::analyzer::AllocSizeVariable*, ikos::analyzer::Variable*, ikos::analyzer::Variable*, ikos::core::machine_int::Interval const&, ikos::analyzer::JsonDict&) + 1264
23 ikos-analyzer            0x0000000100736b68 ikos::analyzer::BufferOverflowChecker::check_mem_access(ikos::ar::Statement*, ikos::ar::Value*, ikos::ar::Value*, ikos::analyzer::Result, ikos::core::exception::ExceptionDomain<ikos::core::memory::PolymorphicDomain<ikos::analyzer::Variable*, ikos::analyzer::MemoryLocation*> >) + 2156
24 ikos-analyzer            0x00000001007360b8 ikos::analyzer::BufferOverflowChecker::check(ikos::ar::Statement*, ikos::core::exception::ExceptionDomain<ikos::core::memory::PolymorphicDomain<ikos::analyzer::Variable*, ikos::analyzer::MemoryLocation*> > const&, ikos::analyzer::CallContext*) + 144
25 ikos-analyzer            0x00000001006000d8 ikos::analyzer::value::interprocedural::sequential::FunctionFixpoint::run_checks() + 1020
26 ikos-analyzer            0x00000001005fe1d8 ikos::analyzer::value::interprocedural::sequential::Analysis::run() + 3484
27 ikos-analyzer            0x000000010052aea8 main + 6168
28 dyld                     0x000000018e4fd0e0 start + 2360
ikos: error: exited with signal SIGTRAP
```

**Comment by ivanperez-keera** [comment]
@mkinney We've just announced a release candidate for IKOS 3.2, and we've included instructions to install it with homebrew. See https://github.com/NASA-SW-VnV/ikos/discussions/253 for details.

@CIPop Can you also try IKOS 3.2-rc1 and, if the issue persists, file a report directly at https://github.com/NASA-SW-VnV/ikos/issues?

Thanks!

**Comment by ivanperez-keera** [comment]
Closed by the solution to #9.
